subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -396,41 +489,104 @@ private void parseData(String data) {
-      System.exit(0);
     }
   }
 
+  private HashMap dataToMap(String[] expected_key, String[] data_array) {
+    HashMap<String, String> hashMap = new HashMap<String, String>();
+    for (int i = 0; i < expected_key.length; i++) {
+      hashMap.put(expected_key[i], data_array[i]);
+    }
+    return hashMap;
+  }
+
+  private String trashLagLines(String data, int index, int lineIndex) {
+    byte[] dataBytes = data.getBytes();
+    int counter = 0;
+    for (int i = 0; i < index; i++) {
+      if (dataBytes[i] == '\n') {
+        counter++;
+      }
+    }
+    if(counter > 0) {
+      counter -= lineIndex;
+    }
+    for (int i = counter; i > 0; i--) {
+      data = trash_line(data, lineIndex);
+    }
+    return data;
+  }
+
+  private String poeNormalizeData(String data) {
+    data = trashLagLines(data, data.indexOf("Interface"), 0);
+    data = trashLagLines(data, data.indexOf("port"), 1);
+    return data;
+  }
+
   private void parseRequestFlag(String data, int requestFlag) {
-    switch (requestFlag) {
-      case 2:
-        // parse show interface
-        login_report += "show interface:\n";
-        parse_packet(data, show_interface_port_expected, show_interface_port_pointers);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 3:
-        // parse show platform
-        login_report += "\nshow platform:\n";
-        parse_packet(data, show_platform_port_expected, show_platform_port_pointers);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 4:
-        // parse show run
-        data = cleanShowRunData(data);
-        login_report += "\n" + data;
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 5:
-        // parse show stack
-        login_report += "\nshow stack:\n";
-        parse_inline(data, show_stack_expected, show_stack_pointers, show_stack_data);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
+    try {
+      switch (requestFlag) {
+        case 2:
+          // parse show interface
+          login_report += "show interface:\n";
+          parse_packet(data, show_interface_port_expected, show_interface_port_pointers);
+          interface_map = dataToMap(interface_expected, show_interface_data);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 3:
+          // parse show platform
+          login_report += "\nshow platform:\n";
+          parse_packet(data, show_platform_port_expected, show_platform_port_pointers);
+          platform_map = dataToMap(platform_expected, show_platform_data);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 4:
+          // parse show power-inline
+          login_report += "\nshow power-inline:\n";
+          if (data.contains("Power-inline is disabled")) {
```

Comment:
@ewRedstone 

The test can never be skipped because of this - even if:

```
"poe" : {
   "enabled" : false
}
```

...in the module_config.json, line 551 re-enables it and the poe tests run anyway - we should expect them to 'skip', no?